client/app/order-panel/order-panel.component.ts

```diff
@@ -82,17 +89,81 @@ export class OrderPanelComponent implements OnInit {
         if (this.discountForm === undefined) {
             return;
         }
-        const discountValue = this.discountForm.value.discount;
-        this.dicountedValue = (discountValue / 100) * this.totalNoDiscount;
 
-        this.totalWithDiscount = this.totalNoDiscount - this.dicountedValue;
+        if (this.order.discount !== undefined && this.discountForm.value.discount === 0) {
```

Comment:
Is there a reason to store the discounted value outside of `this.order.discount`?